cloud/openstack/os_image.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+# Copyright (c) 2013, Benno Joy <benno@ansible.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+#TODO(mordred): we need to support "location"(v1) and "locations"(v2)
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+DOCUMENTATION = '''
+---
+module: os_image
+short_description: Add/Delete images from OpenStack Cloud
+extends_documentation_fragment: openstack
+description:
+   - Add or Remove images from the OpenStack Image Repository
+options:
+   name:
+     description:
+        - Name that has to be given to the image
+     required: true
+     default: None
+   disk_format:
+     description:
+        - The format of the disk that is getting uploaded
+     required: false
+     default: qcow2
+   container_format:
+     description:
+        - The format of the container
+     required: false
+     default: bare
+   owner:
+     description:
+        - The owner of the image
+     required: false
+     default: None
+   min_disk:
+     description:
+        - The minimum disk space required to deploy this image
+     required: false
+     default: None
+   min_ram:
+     description:
+        - The minimum ram required to deploy this image
+     required: false
+     default: None
+   is_public:
+     description:
+        - Whether the image can be accessed publicly. Note that publicizing an image requires admin role by default.
+     required: false
+     default: 'yes'
+   filename:
+     description:
+        - The path to the file which has to be uploaded
+     required: false
+     default: None
+   ramdisk:
+     descrption:
+        - The name of an existing ramdisk image that will be associated with this image
+     required: false
+     default: None
+   kernel:
+     descrption:
+        - The name of an existing kernel image that will be associated with this image
+     required: false
+     default: None
+   properties:
+     description:
+        - Additional properties to be associated with this image
+requirements: ["shade"]
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+'''
+
+EXAMPLES = '''
+# Upload an image from an HTTP URL
```

Comment:
This example says from an http url, but I don't see a URL provided in the example, instead I see a file.